network/codec/cbor/codec.go

```diff
@@ -38,31 +40,53 @@ func (c *Codec) NewDecoder(r io.Reader) network.Decoder {
 func (c *Codec) Encode(v interface{}) ([]byte, error) {
 
 	// encode the value
-	data, code, err := v2envEncode(v, ":wire<1(cbor)")
+	what, code, err := v2envelopeCode(v)
 	if err != nil {
-		return nil, fmt.Errorf("could not encode envelope: %w", err)
+		return nil, fmt.Errorf("could not determine envelope code: %w", err)
 	}
 
-	// encode the envelope
-	//bs := binstat.EnterTime(binstat.BinNet + ":wire<2(cbor)envelope2payload")
-	data = append(data, code)
-	//binstat.LeaveVal(bs, int64(len(data)))
+	// NOTE: benchmarking shows that prepending the code and then using
+	//       .NewEncoder() to .Encode() is the fastest.
 
-	return data, nil
+	// encode / append the envelope code
+	//bs1 := binstat.EnterTime(binstat.BinNet + ":wire<1(cbor)envelope2payload")
+	var data bytes.Buffer
+	data.WriteByte(code)
+	//binstat.LeaveVal(bs1, int64(data.Len()))
+
+	// encode the payload
+	//bs2 := binstat.EnterTime(fmt.Sprintf("%s%s%s:%d", binstat.BinNet, ":wire<2(cbor)", what, code)) // e.g. ~3net::wire<1(cbor)CodeEntityRequest:23
+	encoder := cborcodec.EncMode.NewEncoder(&data)
+	err = encoder.Encode(v)
+	//binstat.LeaveVal(bs2, int64(data.Len()))
+	if err != nil {
+		return nil, fmt.Errorf("could not encode cbor payload of type %s: %w", what, err)
```

Comment:
will the `what` here dump the whole payload? should we instead dump just `code`?

Reply:
@vishalchangrani thanks for the comment. `what` is actually just the human readable text name for `code`. However, I updated the error message to contain both so it is clearer. So the payload is never dumped. Is the new error message clearer now?